grammar.js

```diff
@@ -343,12 +344,58 @@ module.exports = grammar({
     },
 
     // LiteralString
-    string: ($) =>
+    string: $ => choice($._string_literal, $._multiline_string_literal),
```

Comment:
This is a breaking change and I want to discuss it with https://github.com/nvim-treesitter/nvim-treesitter people first.

Can we please do it in a separate PR? That way the fixes for `number` won't be blocked by that discussion.

Reply:
I've been putting of https://github.com/MunifTanjim/tree-sitter-lua/issues/9 all this time because I didn't want to break stuffs. Now that one of the core maintainers of `nvim-treesitter` is interested, it would be nice to make it happen! 😃

I'd still want to discuss it before making a sudden change, because it will also affect `neovim` core and other treesitter plugins. I don't want to break a lot of peoples editor without proper migration plan. Let's discuss it here and come up with a plan? https://github.com/nvim-treesitter/nvim-treesitter/discussions/4706

Reply:
Sure...but since we're not fundamentally changing any rules (only modifying the internals of string), just adding a capture for `@string.escape` (like I have in highlights.scm) would fix it.